### PR TITLE
feat/CTA-13_Delete_Quote_Request

### PR DESCRIPTION
**Pull request by dazhuox**
JIRA: https://champlainsaintlambert.atlassian.net/browse/CTA-13?atlOrigin=eyJpIjoiNGRlOTA5NmM3M2E4NDdjMWE5NzZhNDllN2YwZmI5MDUiLCJwIjoiaiJ9
## Context:
This pull request is a feature addition for deleting a QuoteRequest, both on the admin side and the client side.
## Changes
- Backend methods for Delete request
- Backend testing relevant for jacoco coverage
- Frontend update to display delete button and functions
- Frontend testing with cypress
- Fixes for GetById
## Before and After UI (Required for UI-impacting PRs)
Before:
image

After:
image

## Linked pull requests (Optional)
[- GetById Pull request link](https://github.com/ebond07/CoupeTonArbre/pull/38)

### CODE REVIEW

File: coupetonarbrebackend/src/main/java/com/example/coupetonarbrebackend/configuration/security/SpringSecurityConfigBean.java [hunk 1]
```diff
@@ -91,16 +91,14 @@ public SecurityFilterChain configure(HttpSecurity http) throws Exception {
                                 new AntPathRequestMatcher("/quoteRequests/admin", HttpMethod.GET.toString()),
                                 new AntPathRequestMatcher("/quoteRequests/{quoteRequestId}", HttpMethod.GET.toString()),
                                 new AntPathRequestMatcher("/quoteRequests/{quoteRequestId}/client",HttpMethod.GET.toString()),
+                                new AntPathRequestMatcher("/quoteRequests/{quoteRequestId}/admin",HttpMethod.GET.toString()),
```
Comment (Matthewtan9) [inline coupetonarbrebackend/src/main/java/com/example/coupetonarbrebackend/configuration/security/SpringSecurityConfigBean.java:94]: Next time i feel like the endpoints should be the admin/clients first then the specific client at then end.

Comment (dazhuox) [reply coupetonarbrebackend/src/main/java/com/example/coupetonarbrebackend/configuration/security/SpringSecurityConfigBean.java:94]: yea agreed, i saw some weird stuff going on with a lot of our url's, we should fix them all next sprint

**Review by drallo22 — APPROVED** [review]
Looks good!

**Review by ebond07 — APPROVED** [review]
Everything seems to be in order, good job.

**Pull request merged**